src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,224 @@
+                    methodId,
+                    storeConfig.storeProfile.storeLanguage,
+                    this._paymentMethod.config.testMode,
+                    onPaymentSelect
+                );
+            })
+        .then(() => super.initialize(options));
+    }
+
+    execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+
+        if (!payment || !options) {
+            throw new InvalidArgumentError('Unable to proceed because the customer has not signed into their ChasePay wallet account or selected a payment option.');
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const { initializationData } = this._paymentMethod;
+        const riskToken = (payment.gatewayId === 'wepay') ? this._wepayRiskClient.getRiskToken() : undefined;
+        const paymentData = this._mapCardData(initializationData, riskToken);
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
+            .then(() =>
+                this._store.dispatch(this._paymentActionCreator.submitPayment({
+                    ...payment,
+                    paymentData,
+                }))
+            );
+    }
+
+    private _displayChasepayLightbox(paymentMethod?: PaymentMethod | undefined): Promise<any> {
+        if (!paymentMethod || !paymentMethod.initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+        const { initializationData } = paymentMethod;
+
+        return this._chasePayScriptLoader.load(paymentMethod.config.testMode)
+            .then(({ ChasePay }) => {
+                ChasePay.showLoadingAnimation();
+                ChasePay.startCheckout(initializationData.digitalSessionId);
+            });
+    }
+
+    private _mapCardData(initializationData: any, riskToken?: string | undefined): CryptogramInstrument {
+        let payload: CryptogramInstrument = {
+            cryptogramId: initializationData.paymentCryptogram,
+            eci: initializationData.eci,
+            transactionId: btoa(initializationData.reqTokenId),
+            ccExpiry: {
+                month: initializationData.expDate.toString().substr(0, 2),
+                year: initializationData.expDate.toString().substr(2, 2),
+            },
+            ccNumber: initializationData.accountNum,
+            accountMask: initializationData.accountMask,
+        };
+
+        if (riskToken) {
+            payload = {
+                ...payload,
+                extraData: { riskToken },
+            };
+        }
+        return payload;
+    }
+
+    private _setChasePayConfiguration(methodId: string, storeLanguage: string, isTestMode: boolean | undefined, onPaymentSelect: any): Promise<void> {
```

Comment:
Why does `onPaymentSelect` have `any` type?

Reply:
Because we send the `onPaymentSelect` from`ng-checkout` when we initialize the strategy, so, we don't know which type will it be.

Reply:
Yes you do. You defined it yourself. If you look at `ChasePayInitializeOptions`, you should see that `onPaymentSelect` is defined as `() => void`

Reply:
Ok, got it.